### PR TITLE
REST API returns unknown transaction type

### LINKED ISSUE #8318: REST API returns UNKNOWN transaction type for TokenNftsUpdateTransaction
### Description

/api/v1/transactions/{tx-id} returns:
`"name": "UNKNOWN"`
for a TokenNftsUpdateTransaction

### Steps to reproduce

https://testnet.mirrornode.hedera.com/api/v1/transactions/0.0.2221463-1715863937-138543504

### Additional context

_No response_

### Hedera network

testnet

### Version

 0.105.0-rc1 

### Operating system

None

### PR DESCRIPTION
**Pull request by jascks**
**Description**:

REST API returns `UNKNOWN` transaction type for the new `TokenNftsUpdateTransaction`.

- Add transaction type to `transactiontype.js`.
- Add transaction type to OpenAPI spec enum.
- Add transaction type to Rosetta API.
- Add missing transaction result codes.
- Update go modules.

**Related issue(s)**:

Fixes #8318

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rosetta/app/persistence/transaction_test.go [hunk 1]
```diff
@@ -271,7 +271,7 @@ func TestCategorizeHbarTransfers(t *testing.T) {
 }
 
 func TestGeneralOperationStatus(t *testing.T) {
-	assert.Equal(t, "GENERAL_ERROR", types.GetTransactionResult(400))
+	assert.Equal(t, "GENERAL_ERROR", types.GetTransactionResult(10000))
```
Comment (steven-sheehy) [inline hedera-mirror-rosetta/app/persistence/transaction_test.go:274]: 400 is getting close to actual transaction result ordinal, so increase it to be safe

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by mgoelswirlds — APPROVED** [review]
it is a minor thing but graphql also has a reference to the enums in common.graphqls, so might make sense to add it there also.
Otherwise LGTM.

**Comment by steven-sheehy** [comment]
> it is a minor thing but graphql also has a reference to the enums in common.graphqls, so might make sense to add it there also. Otherwise LGTM.

Good catch. Will address in my follow up.

**Pull request merged**